### ISSUE TITLE
Notebook prefix is incorrectly removed when rendering URLs

### ISSUE THREAD

**Issue by bkowshik** [fastai/nbdev#1194]
I am building the `h3-landmarks` package using `nbdev` and while looking at the initial documentation generated, I see that the URL of the documentation strips numbers from it (`h3_landmarks.html` becomes `hlandmarks.html`), thus making the URL not very logical like we see in this example.

- Actual URL: https://bkowshik.github.io/h3-landmarks/hlandmarks.html
- Expected URL: https://bkowshik.github.io/h3-landmarks/h3_landmarks.html

See the same from the logs of `nbdev_preview` as well when I try it locally.
```
GET: /
GET: /hlandmarks.html
```

## References
- Medium page on why this library: https://bkowshik.medium.com/h3-landmarks-bce6cad971f9
- [WIP] Github repository of the library: https://github.com/bkowshik/h3-landmarks

**Comment by bkowshik** [comment]
Having a similar looking issue with the build on Github failing when the folder name is a number.
Ref: https://github.com/bkowshik/bbmp/actions/runs/3400671873/jobs/5655253340

<img width="1792" alt="Screen Shot 2022-11-05 at 9 11 50 PM" src="https://user-images.githubusercontent.com/2899501/200128117-fc539f07-d4b7-4c95-a0fb-bfe18bac9033.png">

```bash
ERROR: Validation of metadata file /home/runner/work/bbmp/bbmp/_proc/sidebar.yml failed.

In file sidebar.yml
(line 6, columns 18--22) Key section has value 2022, which must be a string
5:       - 00_bbmp.ipynb
6:       - section: 2022
                    ~~~~
7:         contents:
✖ The value 2022 is number.
ℹ The error happened in location website:sidebar:contents:2:section.
```

**Comment by seeM** [comment]
I think you can fix "The value 2022 is number" error by changing the line to `- section: '2022'`.

The second is fixed by #1210 :)

**Comment by jph00** [comment]
I'm not sure that #1210 fixes it correctly so reopening this until that's confirmed.

**Comment by seeM** [comment]
Closed by #1218
